wappuapp-adminpanel/src/actions/auth.js

```diff
@@ -0,0 +1,76 @@
+import axios from 'axios';
+
+import History from '../history';
+
+import { AUTH_USER, UNAUTH_USER, AUTH_ERROR, GET_PROTECTED_DATA } from './types';
+
+const Backend_URL = 'http://localhost:3001/api';
+
+export const login = ({ username, password }) => {
+  return dispatch => {
+    axios
+      .post(
+        `${Backend_URL}/login`,
+        { username: username, password: password },
+        { headers: { 'Content-Type': 'application/json', 'x-user-uuid': 'web' } }
+      )
+      .then(response => {
```

Comment:
.then sijaan ois sika siistiä jos käytettäis async // await ku create-react-app sitä niin kivasti tukee, virheentarkistelu sit menee kivasti try { await juttu } catch (err) { lol }

Reply:
olen samaa mieltä. sen verran monimutkainen tehtävä oli kuitenkin näin osaamattomana etten tarkoituksellakaan halunnut tehdä tästä liian hankalaa tässä vaiheessa

Reply:
async awaittiin tutustumisen jälkeen sillä tulee hitosti helpompaa koodia ku näillä perinteisemmillä promiseilla

Reply:
Tässäkin se ois kivasti
`const response = await axios.post(paramit);
dispatch(paramit);
localStorage.setItem('token', response.data.token);`